piptools/utils.py

```diff
@@ -522,3 +531,129 @@ def parse_requirements_from_wheel_metadata(
             markers=parts.markers,
             extras=parts.extras,
         )
+
+
+def determine_config_file(
+    ctx: click.Context, param: click.Parameter, value: str | None
+) -> Path | None:
+    """
+    Returns the path to the config file with defaults being used, or `None` if no such file is
+    found.
+
+    Defaults for `click.Command` parameters should be override-able in a config file. `pip-tools`
+    will use the first file found, searching in this order: an explicitly given config file, a
+    `.pip-tools.toml`, a `pyproject.toml` file. Those files are searched for in the same directory
+    as the requirements input file.
+    """
+    if value is None:
+        config_file = select_config_file(ctx.params.get("src_files", ()))
+        if config_file is None:
+            return None
+    else:
+        config_file = Path(value)
+
+    if config := parse_config_file(config_file):
```

Comment:
For some reason I thought we already have access to walrus... Python 3.7 is going EOL in 1.5 months so we'll have to wait.
```suggestion
    config = parse_config_file(config_file)
    if config:
```